### PR TITLE
Add default headers from the client config

### PR DESCRIPTION
**Pull request by rpatali**
Client config can specific a dict of default headers that should be
added to every request from that client.

### CODE REVIEW

File: runtime/client_http_request_test.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func TestMakingClientWriteJSONWithBadJSON(t *testing.T) {
 		"clientID",
 		[]string{"DoStuff"},
 		"/",
+		map[string]string{},
```
Comment (oliviazhang0809) [inline runtime/client_http_request_test.go]: can we have test when header is not empty and verify it's being correctly passed in and set?

File: runtime/client_http_response_test.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ func TestReadAndUnmarshalNonStructBody(t *testing.T) {
 		"bar",
 		[]string{"echo"},
 		baseURL,
+		map[string]string{},
```
Comment (oliviazhang0809) [inline runtime/client_http_response_test.go]: same for here

File: runtime/http_client.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ import (
 type HTTPClient struct {
 	Client  *http.Client
 	BaseURL string
+	Headers map[string]string
```
Comment (ChuntaoLu) [inline runtime/http_client.go]: Let's call it `DefaultHeaders` to be consistent with the config filed name.

File: examples/example-gateway/build/clients/bar/bar.go [hunk 4]
```diff
@@ -208,6 +208,10 @@ func NewClient(
 	port := deps.Default.Config.MustGetInt("clients.bar.port")
 	baseURL := fmt.Sprintf("http://%s:%d", ip, port)
 	timeout := time.Duration(deps.Default.Config.MustGetInt("clients.bar.timeout")) * time.Millisecond
+	defaultHeaders := make(map[string]string)
+	if deps.Default.Config.ContainsKey("clients.bar.defaultHeaders") {
```
Comment (cl1337) [inline examples/example-gateway/build/clients/bar/bar.go]: why not always generate `defaultHeaders` empty map in config by default? :)

Comment (ChuntaoLu) [reply examples/example-gateway/build/clients/bar/bar.go]: config file is not necessarily generated. From zanzibar perspective, we assume config files will be read/written by humans, in which sense less clutter is good.

Comment (cl1337) [reply examples/example-gateway/build/clients/bar/bar.go]: make sense, for edge-gateway tho is always gen-ed from ui

File: runtime/client_http_request_test.go (hunk 1 above)
Comment (rpatali) [reply runtime/client_http_request_test.go]: just fyi, i added the test line #134
this test and the next are testing bad input so the request is never sent to the server.

File: runtime/client_http_request_test.go [hunk 5]
```diff
@@ -110,7 +112,7 @@ func TestMakingClientWriteJSONWithBadHTTPMethod(t *testing.T) {
 	assert.Len(t, logs["Could not create outbound request"], 1)
 }
 
-func TestMakingClientCalLWithHeaders(t *testing.T) {
```
Comment (cl1337) [inline runtime/client_http_request_test.go]: lol

**Pull request merged**